panos_upgrade_assurance/check_firewall.py

```diff
@@ -870,6 +864,185 @@ def check_ssl_cert_requirements(self, rsa: dict = {}, ecdsa: dict = {}) -> Check
+
+    def _calculate_time_distance(self, schedule_type: str, schedule: dict) -> (int, str):
+        """A method that calculates the time distance between two `datetime` objects.
+
+        :::note
+        This method is used only by [`CheckFirewall.check_scheduled_updates()`](/panos/docs/panos-upgrade-assurance/api/check_firewall#checkfirewallcheck_scheduled_updates) method and it expects some information
+        to be already available.
+        :::
+
+        # Parameters
+
+        schedule_type (str): A schedule type returned by PanOS, can be one of: `every-*`, `hourly`, `daily`, `weekly`,
+            `real-time`.
+        schedule (dict): Value of the `recurring` key in the API response, see [`FirewallProxy.get_update_schedules()`](/panos/docs/panos-upgrade-assurance/api/firewall_proxy#firewallproxyget_update_schedules)
+            documentation for details.
+
+        # Raises
+
+        MalformedResponseException: Thrown then the `schedule_type` is not recognizable.
+
+        # Returns
+
+        tuple(int, str): A tuple containing the calculated time difference (in minutes) and human-readable description.
+
+        """
+        time_distance = 0
+        details = "unsupported schedule type"
+
+        if schedule_type == "daily":
+            occurrence = schedule["at"]
+            next_occurrence = datetime.strptime(f"{str(self._mp_now.date())} {occurrence}", "%Y-%m-%d %H:%M")
+
+            if self._mp_now > next_occurrence:
+                next_occurrence = next_occurrence + timedelta(days=1)
+            diff = next_occurrence - self._mp_now
+            time_distance = floor(diff.total_seconds() / 60)
+            details = f"at {next_occurrence.time()}"
+
+        elif schedule_type == "hourly":
+            time_distance = 60
+            details = "every hour"
+        elif schedule_type == "weekly":
+            occurrence_time = schedule["at"]
+            occurrence_day = schedule["day-of-week"]
+            occurrence_wday = time.strptime(occurrence_day, "%A").tm_wday
+            now_wday = self._mp_now.weekday()
+
+            diff_days = (0 if occurrence_wday >= now_wday else 7) + occurrence_wday - now_wday
+            next_occurrence_date = (self._mp_now + timedelta(days=diff_days)).date()
+            next_occurrence = datetime.strptime(f"{str(next_occurrence_date)} {occurrence_time}", "%Y-%m-%d %H:%M")
+
+            if self._mp_now > next_occurrence:
+                next_occurrence = next_occurrence + timedelta(days=7)
+            diff = next_occurrence - self._mp_now
+            time_distance = floor(diff.total_seconds() / 60)
+            details = f"in {str(diff).split('.')[0]}"
+
+        elif schedule_type.split("-")[0] == "every":
+            if schedule_type.split("-")[1] == "min":
+                time_distance = 1
+                details = "every minute"
+            elif schedule_type.split("-")[1] == "hour":
+                time_distance = 60
+                details = "hourly"
```

Comment:
I think it would be good to return the same text "every hour" in details as in the above condition in line 932.

```
        elif schedule_type == "hourly":
            time_distance = 60
            details = "every hour"
```

Reply:
👍🏻